recipes-extended/xen/files/libxl-linux-stubdom-replace-disk-with-initramfs.patch

```diff
@@ -0,0 +1,170 @@
+SHORT DESCRIPTION:
+################################################################################
+Use an initramfs instead of a disk for Linux stubdomains created by LibXL
+
+################################################################################
+LONG DESCRIPTION:
+################################################################################
+As implemented by the original RFC, LibXL expects Linux stubdomains to use a
+disk, with a qemu qdisk backend in dom0.
+This is not desired in OpenXT, instead we pack the whole rootfs as an initramfs.
+
+################################################################################
+CHANGELOG
+################################################################################
+Authors:
+Martin Osterloh <osterlohm@ainfosec.com>
+Jed Lejosne <lejosnej@ainfosec.com>
+
+################################################################################
+REMOVAL
+################################################################################
+Probably not desired.
+
+################################################################################
+UPSTREAM PLAN
+################################################################################
+This is specific to OpenXT, but could be upstreamed as an option.
+
+################################################################################
+INTERNAL DEPENDENCIES
+################################################################################
+libxl-RFC-*.patch
+
+################################################################################
+PATCHES
+################################################################################
+Index: xen-4.6.1/tools/libxl/libxl_dm.c
+===================================================================
+--- xen-4.6.1.orig/tools/libxl/libxl_dm.c
++++ xen-4.6.1/tools/libxl/libxl_dm.c
+@@ -1181,7 +1181,7 @@ static int libxl__build_device_model_arg
+                     if (b_info->stubdomain_version == LIBXL_STUBDOMAIN_VERSION_LINUX)
+                         drive = libxl__sprintf
+                                 (gc, "file=%s%c,if=ide,index=%d,media=disk,cache=writeback,format=%s",
+-                                 "/dev/xvd", 'a' + disk, disk, "host_device");
++                                 "/dev/xvd", 'a' + disk, disk, format);
+                     else
+                         drive = libxl__sprintf
+                                 (gc, "file=%s,if=ide,index=%d,media=disk,format=%s,cache=writeback",
+@@ -1384,7 +1384,6 @@ void libxl__spawn_stub_dm(libxl__egc *eg
+     char **args;
+     struct xs_permissions perm[2];
+     xs_transaction_t t;
+-    libxl_device_disk disk_stub;
+ 
+     /* convenience aliases */
+     libxl_domain_config *const dm_config = &sdss->dm_config;
+@@ -1473,20 +1472,12 @@ void libxl__spawn_stub_dm(libxl__egc *eg
+         stubdom_state->pv_ramdisk.path = "";
+         break;
+     case LIBXL_STUBDOMAIN_VERSION_LINUX:
+-        libxl_device_disk_init(&disk_stub);
```

Comment:
Unclear, I believe this was something Martin figured out.  Maybe we could ask him to clarify.